controllers/serviceinstance_controller.go

```diff
@@ -60,19 +65,20 @@ func (r *ServiceInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	if btpOperator == nil {
 		return ctrl.Result{}, nil
 	}
-	if btpOperator.IsReasonStringEqual(string(ServiceInstancesAndBindingsNotCleaned)) {
-		return ctrl.Result{}, r.UpdateBtpOperatorStatus(ctx, btpOperator, types.StateDeleting, HardDeleting, "BtpOperator is to be deleted")
+	if btpOperator.IsReasonStringEqual(string(conditions.ServiceInstancesAndBindingsNotCleaned)) {
+		return ctrl.Result{}, r.UpdateBtpOperatorStatus(ctx, btpOperator, types.StateDeleting, conditions.HardDeleting, "BtpOperator is to be deleted")
 	}
 
 	return ctrl.Result{}, nil
 }
 
-func (r *ServiceInstanceReconciler) UpdateBtpOperatorStatus(ctx context.Context, cr *v1alpha1.BtpOperator, newState types.State, reason Reason, message string) error {
+func (r *ServiceInstanceReconciler) UpdateBtpOperatorStatus(ctx context.Context, cr *v1alpha1.BtpOperator, newState types.State, reason conditions.Reason, message string) error {
 	cr.Status.WithState(newState)
-	newCondition := ConditionFromExistingReason(reason, message)
+	newCondition := conditions.ConditionFromExistingReason(reason, message)
 	if newCondition != nil {
-		SetStatusCondition(&cr.Status.Conditions, *newCondition)
+		conditions.SetStatusCondition(&cr.Status.Conditions, *newCondition)
 	}
+	r.metrics.IncreaseReasonCounter(reason)
```

Comment:
I am not sure if using the same `metrics` might block both reconciliations. Is it possible to populate our controllers with separate metrics instances and avoid a mutex at all?